### PR TITLE
add support to install roles in EE and EE creation in awx

### PR DESCRIPTION
**Pull request by jangel97**
Add support for to install roles in EE

### CODE REVIEW

**Review by djdanielsson — CHANGES_REQUESTED** [review]
I am not sure we want to add this here, we already have our controller config collection that has this ability. I would argue that is where you would need to add in code the EE that you want in your controller and not here but I am open to hear others opinion on this. At minimum see my other comment because that change would be required before approval.

**Comment by jangel97** [comment]
Hi @djdanielsson,

I am using the role ee_builder in my workflow to build Execution Environments in Ansible Automation Platform. It felt like a natural step to me that once you build and push the container image, the next step is to create the Execution Environment object on the controller. That is why it made sense for me to add it.

What about the addition of the roles entry in the requirements.yml?

File: roles/ee_builder/tasks/00_build_ee.yml
Comment (jangel97) [reply roles/ee_builder/tasks/00_build_ee.yml]: I removed it in commit 2bd9fe480f3

**Comment by djdanielsson** [comment]
So I will wait for some of the other maintainers @sean-m-sullivan @Tompage1994  to voice their thoughts on the EE object in controller but as for the roles I think that is fine. I am not sure how often that would be used but I off hand do not know of a reason not to add it. You would need to add the role option to the readme and then hopefully we will have a decision for you about the adding to controller part.

**Comment by jangel97** [comment]
Hi,

Sounds good! I updated the Readme. If everyone agrees on the change, please, wait until I run few more tests in my environment.

Also, let me know if you have any other concern.

Thanks!

**Comment by branic** [comment]
My 2 cents are that pushing the EE to a registry doesn't fit with this collection because someone may not always want to publish the EE directly to Controller. A user may want to publish to Automation Hub instead. So I think the publishing of roles is better left to the existing functionality in the [controller_configuration](https://github.com/redhat-cop/controller_configuration) and [ah_configuration](https://github.com/redhat-cop/ah_configuration) collections or user provided automation to push to some other registry. A playbook could easily utilize this collection to build the EE and the controller_configuration collection to push to controller if that is the desired publication path.

**Comment by branic** [comment]
The above being said. I do think the support to allow this collection to add roles to an EE is valuable.

**Comment by sean-m-sullivan** [comment]
At least to me, I've always seen the redhat_cop collections as a suite of collections that are built to interact with their individual components. and these are individual roles packaged for consumption. So while this role is built to build the EE and interact with builder, and do other functions, the redhat_cop wraparound for the controller.execution environment is the way to do this invoked afterward with the full definition, as this PR its missing A TON of the options available for that. Like this:
```
  roles:
    - redhat_cop.ee_utilities.ee_builder
    - redhat_cop.controller_configuration.execution_environment
    ```
Part of it is not having to update code in 2 places when changes happen, and to put things together, we could add the controller configuration to the meta requirements, but I'm hesitant to do that.

**Review by djdanielsson — CHANGES_REQUESTED** [review]
Ok so I think it has been decided, let's remove the EE push to controller and only add the roles content with this PR. If you look at Sean's example I think that is the simplest way for you to achieve what you are looking for without changing the current system.

**Comment by jangel97** [comment]
Hi,
Sounds good. I removed the controller EE part. 
Let me know how it looks now.
Thank you!

**Review by djdanielsson — CHANGES_REQUESTED** [review]
So it looks like you removed your readme additions for the role section that need to come back in and then I think we are good to go

**Pull request merged**
